### ISSUE TITLE
Add and handle Process Deleting event

### ISSUE THREAD

**Issue by remcowesterhoud** [camunda/zeebe#13349]
## Description

The process deleting event will mark the process definition as pending deletion.

- Add the `DELETING` intent to the `ProcessIntent`
- Add an `EventApplier` to handle the `DELETING` intent
    - The applier must change the state the `PersistedProcess` to `PENDING_DELETION`
        - Don't forget to change the state in the `CoulmnFamily`, as well as in the cache.
    - This means a new method must be created on the `DbProcessState`
        - Include tests!

**Out of scope**
- Writing the event will happen in a different issue

**Blocked by**
#13348
